docs/framework/upgrade.md

```diff
@@ -0,0 +1,84 @@
+# Upgrade Reference
+
+!!! note "Scope"
+    This document discusses upgrade conventions used in CiviCRM core code. For extensions and modules which integrate with CiviCRM, upgrade conventions are different. See [Create a Module Extension](/extensions/civix.md) for more discussion of the module-focused upgrader.
+
+## Introduction
+
+Within CiviCRM Core code the upgrader classes handle making any necessary database changes either through PHP or SQL code, printing out any messages before or after the upgrade happens. Queuing up tasks for each version upgrade.
```

Comment:
* As an "introduction", this is probably better than the old one - ie it quickly summarizes *what* happens rather than digging into *how*.
 * The last sentence is not a sentence. Kill it?
 * It is useful to break-down the mechanics of the process. I guess the question is whether material from [Introduction](https://wiki.civicrm.org/confluence/display/CRMDOC/Upgrade+Reference#UpgradeReference-Introduction) and [Order of Upgrade Steps](https://wiki.civicrm.org/confluence/display/CRMDOC/Upgrade+Reference#UpgradeReference-OrderofUpgradeSteps) are duplicates?